iothub/device/src/IotHubDeviceClient.cs

```diff
@@ -74,21 +70,19 @@ private IotHubDeviceClient(IotHubConnectionCredentials iotHubConnectionCredentia
             }
 
             // Validate certificates
-            if (IotHubConnectionCredentials.AuthenticationMethod is ClientAuthenticationWithX509Certificate x509CertificateAuth
+            if (IotHubConnectionCredentials?.AuthenticationMethod is ClientAuthenticationWithX509Certificate x509CertificateAuth
                 && x509CertificateAuth.CertificateChain != null)
             {
-                if (_clientOptions.TransportSettings.Protocol != IotHubClientTransportProtocol.Tcp)
+                if (_clientOptions?.TransportSettings?.Protocol != IotHubClientTransportProtocol.Tcp)
```

Comment:
Client options should never be null as it is initialized in base client. Transport settings is required to be not null in ctor. Protocol is enum, not nullable.

Not harmful, but shouldn't be necessary and I'd wonder why.

Reply:
Removed